Tests/MacroToolkitTests/DeclGroupTests.swift

```diff
@@ -0,0 +1,161 @@
+import XCTest
+import SwiftSyntax
+@testable import MacroToolkit
+
+final class DeclGroupTests: XCTestCase {
+    
+    func testStructInitialization() throws {
+        let decl: DeclSyntax = """
+        struct TestStruct { var value: Int }
+        """
+        let structDecl = decl.as(StructDeclSyntax.self)!
+        let testStruct = Struct(structDecl)
+        
+        XCTAssertEqual(testStruct.identifier, "TestStruct")
+        XCTAssertEqual(testStruct.members.count, 1)
+        XCTAssertEqual(testStruct.properties.count, 1)
+    }
+    
+    func testEnumInitialization() throws {
+        let decl: DeclSyntax = """
+        enum TestEnum { case caseOne, caseTwo }
+        """
+        let enumDecl = decl.as(EnumDeclSyntax.self)!
+        let testEnum = Enum(enumDecl)
+        
+        XCTAssertEqual(testEnum.identifier, "TestEnum")
+        XCTAssertEqual(testEnum.members.count, 1)
+        XCTAssertEqual(testEnum.cases.count, 2)
+    }
+    
+    func testClassInitialization() throws {
+        let decl: DeclSyntax = """
+        class TestClass { var value: Int }
+        """
+        let classDecl = decl.as(ClassDeclSyntax.self)!
+        let testClass = Class(classDecl)
+        
+        XCTAssertEqual(testClass.identifier, "TestClass")
+        XCTAssertEqual(testClass.members.count, 1)
+        XCTAssertEqual(testClass.properties.count, 1)
+    }
+    
+    func testActorInitialization() throws {
+        let decl: DeclSyntax = """
+        actor TestActor { var value: Int }
+        """
+        let actorDecl = decl.as(ActorDeclSyntax.self)!
+        let testActor = Actor(actorDecl)
+        
+        XCTAssertEqual(testActor.identifier, "TestActor")
+        XCTAssertEqual(testActor.members.count, 1)
+        XCTAssertEqual(testActor.properties.count, 1)
+    }
+    
+    func testExtensionInitialization() throws {
+        let decl: DeclSyntax = """
+        extension TestStruct { func testMethod() {} }
+        """
+        let extensionDecl = decl.as(ExtensionDeclSyntax.self)!
+        let testExtension = Extension(extensionDecl)
+        
+        XCTAssertEqual(testExtension.identifier, "TestStruct")
+        XCTAssertEqual(testExtension.members.count, 1)
+    }
+    
+    func testDeclGroupInitialization() throws {
+        let structDecl: DeclSyntax = """
+        struct TestStruct { var value: Int }
+        """
+        let structSyntax = structDecl.as(StructDeclSyntax.self)!
+        let structDeclGroup = DeclGroup(structSyntax)
+        
+        switch structDeclGroup {
+        case .struct(let testStruct):
+            XCTAssertEqual(testStruct.identifier, "TestStruct")
```

Comment:
I reckon that if this test passes it's highly likely that `testStructInitialization` passes too (since this is kinda acting as a simple integration test). Might be best to reduce overlapping coverage between all of these tests to make them easier to keep in sync.

Reply:
This ensures that the Struct initialization logic is independently verified. This helps catch issues specific to Struct initialization without relying on the broader context. If this passes the `testStructInitialization` will absolutely pass but the converse is not necessarily true.